fitbenchmarking/controllers/matlab_mixin.py

```diff
@@ -53,3 +56,19 @@ def py_to_mat(func, eng):
         eng.evalc('fp.close()')
 
         return eng.workspace['fm']
+
+    def validate(self):
+        """
+        Validates that the provided options are compatible with each other.
+        If there are some invalid options, the relevant exception is raised.
+        """
+        self._validate_problem_format()
+        super().validate()
+
+    def _validate_problem_format(self):
+        """
+        Validates that the problem format is compatible with matlab controllers
+        """
+        if self.problem.format == 'mantid':
+            raise IncompatibleProblemError(
+                'Mantid problems cannot be used with MATLAB controllers.')
```

Comment:
Would it be possible to move this to the base controller and get it to do something similar to `_validate_jacobian` so that this method wouldn't need reimplementing in another controller if we need similar validation in the future?

So for instance could we have an empty list of `INVALID_PROBLEMS` in the base controller, and then in the matlab controller we could have `INVALID_PROBLEMS = ["mantid"]`. Then we could check if the `self.problem.format` is in the `INVALID_PROBLEMS`? A bit like what is done for the jacobians:
https://github.com/fitbenchmarking/fitbenchmarking/blob/780eeaa485480f2bede7ca5eaf7763ddc4240c8c/fitbenchmarking/jacobian/base_jacobian.py#L15

https://github.com/fitbenchmarking/fitbenchmarking/blob/780eeaa485480f2bede7ca5eaf7763ddc4240c8c/fitbenchmarking/jacobian/scipy_jacobian.py#L17

Reply:
Absolutely, I'll update